onnxruntime/core/mlas/lib/fp16_common.h

```diff
@@ -64,6 +64,23 @@ MLAS_FORCEINLINE
 MLAS_FLOAT16X4
 MlasLoadFloat16x4(const _mlas_fp16_* Buffer) { return vreinterpret_f16_u16(vld1_u16(Buffer)); }
 
+MLAS_FORCEINLINE
+MLAS_FLOAT16X4
+MlasLoadPartialFloat16x4(const _mlas_fp16_* Buffer, size_t len)
+{
+    MLAS_FLOAT16X4 Vector = MlasZeroFloat16x4();
+    if ((len & 1) != 0) {
+        Vector = vreinterpret_f16_u16(vld1_lane_u16(Buffer + (len - 1), vreinterpret_u16_f16(Vector), 0));
```

Comment:
if `len` is 3, this should load `Buffer + 3 - 1` into lane 3, right?

Reply:
perhaps something like this would work:
```
// assuming len is 1, 2, or 3
if (len & 2) {  // len is at least 2
  // load first two lanes, can use one vld1_lane_f32
  if (len & 1) {  // len is 3
    // load third lane
  }
} else {  // len is 1
  // load first lane
}

Reply:
For len == 3,
`Vector = vreinterpret_f16_u16(vld1_lane_u16(Buffer + (3 - 1), vreinterpret_u16_f16(Vector), 0)` would give
Mem:   [0, 1, 2, x]
Vector: [2, x, x, x]
Then, `Vector = vreinterpret_f16_f32(vdup_lane_f32(vreinterpret_f32_f16(Vector), 0));` would give
Vector: [2, x, 2, x]
Finally, `Vector = vreinterpret_f16_f32(vld1_lane_f32(reinterpret_cast<const float*>(Buffer), vreinterpret_f32_f16(Vector), 0));` would give
Vector: [0, 1, 2, x]

The added tests covered the three cases that len == 1, 2, 3

Reply:
```
// assuming len is 1, 2, or 3
if (len & 2) {  // len is at least 2
  // load first two lanes, can use one vld1_lane_f32
  if (len & 1) {  // len is 3
    // load third lane
  }
} else {  // len is 1
  // load first lane
}
```
It would have two comparisons when len is 3, or do you think using a switch is better here?

Reply:
ah I see, thanks for explaining it. a switch may make what's going on more obvious but I think it's fine like this.

Reply:
That's a great explanation! might as well add to code as comments.